File: cwclock-mobile/lib/screens/onboarding/connect_screen.dart
import 'package:flutter/material.dart';
import 'package:flutter_riverpod/flutter_riverpod.dart';
import 'package:go_router/go_router.dart';

import '../../providers/locale_provider.dart';
import '../../theme.dart';
import '../../widgets/app_button.dart';
import '../../widgets/app_screen.dart';

/// The onboarding entry point: two ways into the same underlying flow, both
/// ending at SessionNotifier.connect. Neither button navigates directly past
/// onboarding itself - the router's redirect (see router.dart) reacts to the
/// resulting session status change. Ported from
/// src/screens/onboarding/ConnectScreen.js.
class ConnectScreen extends ConsumerWidget {
  const ConnectScreen({super.key});

  @override
  Widget build(BuildContext context, WidgetRef ref) {
    final locale = ref.watch(localeProvider);
    final t = translateWith(locale);

    return AppScreen(
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          SizedBox(height: AppSpacing.of(6)),
          Text(
            t('onboarding.title'),
            style: const TextStyle(fontSize: 24, fontWeight: FontWeight.w700, color: AppColors.text),
          ),
          SizedBox(height: AppSpacing.of(1)),
          Text(t('onboarding.intro'), style: const TextStyle(fontSize: 15, color: AppColors.textMuted)),
          SizedBox(height: AppSpacing.of(4)),
          AppButton(
            title: t('onboarding.scanQr'),
            onPressed: () => context.push('/onboarding/scan-qr'),
            margin: EdgeInsets.only(bottom: AppSpacing.of(2)),
          ),
          AppButton(
            title: t('onboarding.enterManually'),
            variant: AppButtonVariant.secondary,
            onPressed: () => context.push('/onboarding/manual-entry'),
          ),
        ],
      ),
    );
  }
}

File: cwclock-mobile/src/screens/onboarding/ConnectScreen.js
import React from "react";
import { StyleSheet, Text, View } from "react-native";
import Screen from "../../components/Screen";
import Button from "../../components/Button";
import { useI18n } from "../../i18n/I18nContext";
import theme from "../../theme";

// ConnectScreen is the onboarding entry point (see the ai-instruct-105 plan
// - "ask to scan the QR code or the api key"): two ways into the same
// underlying flow, both ending at redux/session/session.actions.js's
// connectApi. Session status (src/redux/session/session.reducer.js) drives
// which screen the root navigator shows next (see src/App.js) - neither
// button here navigates directly past onboarding itself.
const ConnectScreen = ({ navigation }) => {
  const { t } = useI18n();

  return (
    <Screen>
      <View style={styles.header}>
        <Text style={styles.title}>{t("onboarding.title")}</Text>
        <Text style={styles.intro}>{t("onboarding.intro")}</Text>
      </View>
      <Button title={t("onboarding.scanQr")} onPress={() => navigation.navigate("ScanQr")} style={styles.button} />
      <Button
        title={t("onboarding.enterManually")}
        variant="secondary"
        onPress={() => navigation.navigate("ManualEntry")}
      />
    </Screen>
  );
};

const styles = StyleSheet.create({
  header: {
    marginTop: theme.spacing(6),
    marginBottom: theme.spacing(4),
  },
  title: {
    fontSize: 24,
    fontWeight: "700",
    color: theme.color.text,
    marginBottom: theme.spacing(1),
  },
  intro: {
    fontSize: 15,
    color: theme.color.textMuted,
  },
  button: {
    marginBottom: theme.spacing(2),
  },
});

export default ConnectScreen;
